doc-content/shared-kie-docs/src/main/asciidoc/KieServer/controller-java-api-requests-proc.adoc

```diff
@@ -0,0 +1,126 @@
+[id='controller-java-api-requests-proc_{context}']
+= Sending requests with the {CONTROLLER} Java client API
+
+The {CONTROLLER} Java client API enables you to connect to the {CONTROLLER} using REST or WebSocket protocols from your Java client application. You can use the {CONTROLLER} Java client API as an alternative to the {CONTROLLER} REST API to interact with your {KIE_SERVER} templates (configurations), {KIE_SERVER} instances (remote servers), and associated KIE containers (deployment units) in {PRODUCT} without using the {CENTRAL} user interface.
+
+.Prerequisites
+* {KIE_SERVER} is installed and running.
+* The {CONTROLLER} or {HEADLESS_CONTROLLER} is installed and running.
+* You have `rest-all` user role access to the {CONTROLLER} if you installed {CENTRAL}, or `kie-server` user role access to the {HEADLESS_CONTROLLER} installed separately from {CENTRAL}.
+* You have a Java project with {PRODUCT} resources.
+
+.Procedure
+. In your client application, ensure that the following dependencies have been added to the relevant classpath of your Java project:
++
+[source,xml,subs="attributes+"]
+----
+// For remote execution on controller
+<dependency>
+  <groupId>org.kie.server</groupId>
+  <artifactId>kie-server-controller-client</artifactId>
+  <version>${{PRODUCT_INIT}.version}</version>
+</dependency>
+
+// For REST client
+<dependency>
+  <groupId>org.jboss.resteasy</groupId>
+  <artifactId>resteasy-client</artifactId>
+  <version>${resteasy.version}</version>
+</dependency>
+
+// For WebSocket client
+<dependency>
+  <groupId>io.undertow</groupId>
+  <artifactId>undertow-websockets-jsr</artifactId>
+  <version>${undertow.version}</version>
+</dependency>
+
+// For debug logging (optional):
+<dependency>
+  <groupId>ch.qos.logback</groupId>
+  <artifactId>logback-classic</artifactId>
+  <version>${logback.version}</version>
+</dependency>
+----
+ifdef::DM,PAM[]
+. Download the *{PRODUCT} {PRODUCT_VERSION_LONG} Source Distribution* from the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Red Hat Customer Portal] and navigate to `~/{PRODUCT_FILE}-sources/src/droolsjbpm-integration-$VERSION/kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client` to access the {CONTROLLER} Java clients.
+endif::[]
+. In the `../kie/server/controller/client` folder
+ifdef::DROOLS,JBPM,OP[]
+of the Java client API in https://github.com/kiegroup/droolsjbpm-integration/tree/master/kie-server-parent/kie-server-controller/kie-server-controller-client/src/main/java/org/kie/server/controller/client[GitHub]
+endif::[]
+, identify the relevant Java client for the request you want to send, such as `RestKieServerControllerClient` to access client services for {KIE_SERVER} templates and KIE containers in REST protocol.
```

Comment:
I believe that identifying the correct implementation class can be simplified to identifying which communication protocol a user wants to use - REST or WebSocket. After that a user just calls the `newRestClient(...)` or the `newWebSocketClient(...)` method respectively. wdyt @mswiderski?

Reply:
Right, that's what I mean by "identify". It's important for the user to go and see the client services and implementations to become familiar with them as part of this procedure, imo.

Reply:
OK, I understand the intention now. Thanks.